os/drivers/audio/ndp120_voice.c

```diff
@@ -469,8 +467,8 @@ static int ndp120_enqueuebuffer(FAR struct audio_lowerhalf_s *dev, FAR struct ap
 	}
 
 	sq_entry_t *tmp;
-	apb->nbytes = CONFIG_NDP120_BUFFER_SIZE;
-	ndp120_extract_audio(priv, apb->samp, CONFIG_NDP120_BUFFER_SIZE);
+	apb->nbytes = priv->sample_size;
+	ndp120_extract_audio(priv, apb->samp, priv->sample_size);
```

Comment:
```
ndp120_extract_audio(priv, apb);
```
and then ndp120_extract_audio should handle it & fill buffer & update apb->nbytes.